tasks/unit-tests/modules_test.py

```diff
@@ -0,0 +1,81 @@
+import json
+import os
+import subprocess
+import unittest
+from typing import Any, Set
+
+from ..modules import DEFAULT_MODULES
+
+AGENT_MODULE_PATH_PREFIX = "github.com/DataDog/datadog-agent/"
+
+
+class TestModules(unittest.TestCase):
+    def load_go_mod(self, module_path: str) -> Any:
+        """Loads the go.mod file as a JSON object"""
+        go_mod_path = os.path.join(module_path, "go.mod")
+        res = subprocess.run(["go", "mod", "edit", "-json", go_mod_path], capture_output=True)
+        self.assertEqual(res.returncode, 0)
+
+        return json.loads(res.stdout)
+
+    def get_agent_required(self, module: Any) -> Set[str]:
+        """Returns the set of required datadog-agent modules"""
+        self.assertIsInstance(module, dict)
```

Comment:
This assertion is duplicated. We check in `get_agent_required` and `get_agent_replaced`. What do you think about moving into the primary test function?

Reply:
I thought about it and considered they could be called separately, but since they aren't right now anyway I'll update that